### ISSUE TITLE
Slow Launcher drawer

### ISSUE THREAD

**Issue by xi7** [clemensbartz/essential-launcher#94]
i can see icon loading while opening all time, very very slow drawer.

small launcher, but slow !

**Comment by clemensbartz** [comment]
Hello,

Can you please tell me which device you are using and which Android Version this is?

**Comment by xi7** [comment]
> Hello,
> 
> Can you please tell me which device you are using and which Android Version this is?
[recording_20201116_094732.zip](https://github.com/clemensbartz/essential-launcher/files/5547291/recording_20201116_094732.zip)
adnroid 5.0 test with 3 different launcher, always reload icon and slowest each day

**Comment by clemensbartz** [comment]
Thank you for providing this. Can you please also provide the RAM usage for the apps you compared?

**Comment by xi7** [comment]
> Thank you for providing this. Can you please also provide the RAM usage for the apps you compared?

download the video, you can check speed each test.

1gb ram device. all other launcher work normally and fast.

**Comment by xi7** [comment]
![Screenshot_2020-11-16-17-00-13](https://user-images.githubusercontent.com/67331785/99302133-b463a280-282d-11eb-91c6-8cdac27c04b3.png)
![Screenshot_2020-11-16-17-00-44](https://user-images.githubusercontent.com/67331785/99302141-b6c5fc80-282d-11eb-8318-1f6139c67d4b.png)
![Screenshot_2020-11-16-17-01-16](https://user-images.githubusercontent.com/67331785/99302148-b9285680-282d-11eb-8406-eb356a60821f.png)
![Screenshot_2020-11-16-17-02-10](https://user-images.githubusercontent.com/67331785/99302153-ba598380-282d-11eb-9830-924e3ef592f6.png)

**Comment by clemensbartz** [comment]
Thank you very much. Except for Lite Launcher, the RAM usages confirm my expectations. 

The design decision behind Essential Launcher is a trade-off between CPU, RAM and disk usage. All Launcher3-based launchers are using an [IconCache](https://github.com/amirzaidi/Launcher3/blob/o-mr1/src/com/android/launcher3/IconCache.java) (example from one of the forks). I did not want to hold the icons all the time in RAM and therefore give other apps less of it. Essential Launcher does not have such a cache. 

Essential Launcher is loading all icons on demand which depending on the storage provider that is built into the device can take a while. This gives the user the option to open the app without having to wait for the icon to appear. 

On the dock, the icons are loaded with the text, no delay should be observed there. 

I chose this option consciously in order to save space of the app (less code) and not use as much RAM when the application is not running. The assumption is that users put their most used icons in the dock. This way I don't have to use CPU and disk (i. e. battery) on loading and holding most icons in RAM since they are not needed 80 % of the time. 

In summary however, I see no action points as the loading time of all icons of the page is within 1-2 seconds. I agree it is not instant, but in comparison to the development device I am testing this with (Alcatel Pixi 4) your loading times in the video are lightning speed. 

I will take a look at Lite Launcher though and check what they are doing differently.

**Comment by xi7** [comment]
[recording_20201117_082433.zip](https://github.com/clemensbartz/essential-launcher/files/5553128/recording_20201117_082433.zip)

Good Test with Light Launcher v1.2
**(no cache and just 48kb disk / 3mb ram) Load Super fast and instant**

**about  Essential Launcher all time reload until the dock icon**

**Comment by xi7** [comment]
![Screenshot_2020-11-18-19-21-01](https://user-images.githubusercontent.com/67331785/99663558-91134000-2a45-11eb-9b6a-267f4f0ca0f6.jpg)

**3C_Task_Manager_Pro**
